### ISSUE TITLE
[6.0.100-preview.7.21355.1] IHubContext MissingMethodException "get_Clients() not found" in MixCore

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/aspnetcore#34111]
**Application Name**: mixCore
**OS**: Windows 10 RS5
**CPU**: X64
**.NET Build Number**: 6.0.100-preview.7.21355.1

**Verify Scenarios:**
1) Windows 10 RS5 X64 + dotnet-sdk-5.0.204-win-x64 PASS
2) Windows 10 RS5 X64 + dotnet-sdk-6.0.100-preview.6.21351.1  PASS
3) Windows 10 RS5 X64 + 6.0.100-preview.7.21355.1: Fail

**Source Code & App check at** : https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1352071

**Repro steps:**
This is a net5 app, issue occurs when we run against net6-prev7
1)Copy App to machine
2) Run dotnet mixcore.dll
3) Open http://localhost:5000/

**Expected Result:**
No Error

**Actual Result:**
There is error

Error :
```
Occurs in Microsoft.AspNetCore.SignalR.Core.dll

fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.MissingMethodException: Method not found: 'Microsoft.AspNetCore.SignalR.IHubClients Microsoft.AspNetCore.SignalR.IHubContext`1.get_Clients()'.
         at Mix.Cms.Api.Controllers.v1.BaseApiController`1.AlertAsync(String action, Int32 status, String message)
         at Mix.Cms.Api.Controllers.v1.BaseApiController`1.OnActionExecuting(ActionExecutingContext context) in C:\EfCore\mix.core\src\Mix.Cms.Api\Controllers\v1\BaseApiControoler.cs:line 60
         at Microsoft.AspNetCore.Mvc.Controller.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in Microsoft.AspNetCore.Mvc.ViewFeatures.dll:token 0x6000093+0x1c
         at Microsoft.AspNetCore.Mvc.Filters.ControllerActionFilter.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000b41+0x45
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x60009bd+0x16b
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync() in Microsoft.AspNetCore.Mvc.Core.dll:token 0x60009c1+0xa
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a94+0x6a
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a89+0x15
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a80+0x835
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync() in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a7d+0xa
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a8c+0x77
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a8c+0xfb
         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger) in Microsoft.AspNetCore.Routing.dll:token 0x60000ab+0x5e
         at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext) in Swashbuckle.AspNetCore.SwaggerUI.dll:token 0x6000002+0x1ce
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider) in Swashbuckle.AspNetCore.Swagger.dll:token 0x6000009+0x8e
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Authorization.Policy.dll:token 0x6000013+0x16b
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Authentication.dll:token 0x6000049+0x3be
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Diagnostics.dll:token 0x60000aa+0x82
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.MissingMethodException: Method not found: 'Microsoft.AspNetCore.SignalR.IHubClients Microsoft.AspNetCore.SignalR.IHubContext`1.get_Clients()'.
         at Mix.Cms.Api.Controllers.v1.BaseApiController`1.AlertAsync(String action, Int32 status, String message)
         at Mix.Cms.Api.Controllers.v1.BaseApiController`1.OnActionExecuting(ActionExecutingContext context) in C:\EfCore\mix.core\src\Mix.Cms.Api\Controllers\v1\BaseApiControoler.cs:line 60
         at Microsoft.AspNetCore.Mvc.Controller.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in Microsoft.AspNetCore.Mvc.ViewFeatures.dll:token 0x6000093+0x1c
         at Microsoft.AspNetCore.Mvc.Filters.ControllerActionFilter.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000b41+0x45
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x60009bd+0x16b
         at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync() in Microsoft.AspNetCore.Mvc.Core.dll:token 0x60009c1+0xa
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a94+0x6a
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a89+0x15
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a80+0x835
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync() in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a7d+0xa
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a8c+0x77
         at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope) in Microsoft.AspNetCore.Mvc.Core.dll:token 0x6000a8c+0xfb
         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger) in Microsoft.AspNetCore.Routing.dll:token 0x60000ab+0x5e
         at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext) in Swashbuckle.AspNetCore.SwaggerUI.dll:token 0x6000002+0x1ce
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider) in Swashbuckle.AspNetCore.Swagger.dll:token 0x6000009+0x8e
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Authorization.Policy.dll:token 0x6000013+0x16b
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Authentication.dll:token 0x6000049+0x3be
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context) in Microsoft.AspNetCore.Diagnostics.dll:token 0x60000aa+0x82
```
**Findings** :

MissingMethodException in IHubContext
```
 IHubContext<MyHub> _hubContext;
 await _hubContext.Clients.All.SendAsync("receiveMessage", "Test");
```

@dotnet-actwx-bot @dotnet/compat

**Comment by BrennanConroy** [comment]
Can you please provide better repro steps?

>Copy App to machine

I copied the entire folder over (more than 600MB... please don't include bin and obj folders in the future). And built Mix.Cms.Web and ran it, no issues.

**Comment by Junjun-zhao** [comment]
@

> Can you please provide better repro steps?
> 
> > Copy App to machine
> 
> I copied the entire folder over (more than 600MB... please don't include bin and obj folders in the future). And built Mix.Cms.Web and ran it, no issues.

@BrennanConroy  Thanks for looking into this issue. Please just copy the one Under "App" which is the published app and smaller.  The one in "App Source" is the Source code of the app and much larger(We put it there just in case you want to check the source).  We have updated the steps and also please make sure to run the app with SDK 6.0 Prev7.  Please let us know if you have any more question.

**Comment by BrennanConroy** [comment]
How did you generate the publish output? Did you handcraft the mixcore.runtimeconfig.json file?

It looks like:
```
{
  "runtimeOptions": {
    "tfm": "net5.0",
    "framework": {
      "name": "Microsoft.AspNetCore.App",
      "version": "6.0.0-preview.7.21330.3"
    }
  }
}
```

You shouldn't be running a net5.0 app against a 6.0 runtime.

**Comment by Junjun-zhao** [comment]
@BrennanConroy  We updated this file manually to make sure it runs against 6.0. Actually this is net 5.0 app.   We are from AppCompat team and we are testing 2.x, 3.x, 5.x, 6.x apps against latest runtime to find compatibility issues.

**Comment by BrennanConroy** [comment]
Breaking change fixed via https://github.com/dotnet/aspnetcore/pull/34174

**Comment by Junjun-zhao** [comment]
@BrennanConroy  Thanks! Is there a breaking change announcement for this behavior change?

**Comment by BrennanConroy** [comment]
No, because there shouldn't be a breaking change

**Comment by Junjun-zhao** [comment]
Thanks. We will verify it on the latest build and update the status

**Comment by Junjun-zhao** [comment]
Verified and this issue has been fixed on SDK dotnet-sdk-6.0.100-preview.7.21360.1
